proof.go

```diff
@@ -141,6 +141,58 @@ func (proof Proof) IsEmptyProof() bool {
 	return proof.start == proof.end && len(proof.nodes) == 0 && len(proof.leafHash) == 0
 }
 
+// IsValidEmptyRangeProof checks whether the proof is a valid empty range proof.
+func (proof Proof) IsValidEmptyRangeProof(nth *NmtHasher, nID namespace.ID, root []byte, leaves [][]byte, checkNS bool) bool {
+	nIDLen := nID.Size()
+
+	if !proof.IsEmptyProof() || len(leaves) != 0 {
+		return false
+	}
+
+	if !checkNS {
+		return true
+	}
+
+	rootMin := namespace.ID(MinNamespace(root, nIDLen))
+	rootMax := namespace.ID(MaxNamespace(root, nIDLen))
+
+	// empty proofs are always rejected unless 1) nID is outside the range of
+	// namespaces covered by the root 2) the root represents an empty tree, since
+	// it purports to cover the zero namespace but does not actually include
+	// any such nodes
+	return nID.Less(rootMin) || rootMax.Less(nID) || bytes.Equal(root, nth.EmptyRoot())
+}
+
+// ComputeLeafHashes computes the hashes of the given leaves by prepending the namespace ID to each leaf and hashing it using the provided NMT hasher.
+func ComputeLeafHashes(nth *NmtHasher, nid namespace.ID, leaves [][]byte, checkOrAppend bool) ([][]byte, error) {
```

Comment:
Having such` checkOrAppend bool` flags/modes makes the code harder to read and maintain in the future. The user of this method will have to dive into this code and understand what is it for. Also, I would verify leaves every time during computation.

Reply:
Yeah seems like two different semantics. Input is also different: one takes in leaf with prepended ns and another one does not. Maybe just make two different funcs or even extract validatation/prepending from hashing?
Also similar to above might be worth to keep private func for now if it is not meant to be used outside of pkg

Reply:
split into two separate functions.